xbmc/guilib/GUIIncludes.cpp

```diff
@@ -513,13 +513,22 @@ std::string CGUIIncludes::ResolveConstant(const std::string &constant) const
   return StringUtils::Join(values, ",");
 }
 
-std::string CGUIIncludes::ResolveExpressions(const std::string &expression) const
+std::string CGUIIncludes::ResolveExpressions(const std::string &strInput, const std::vector<std::string> &resolved /* = std::vector<std::string>() */) const
 {
-  std::string work(expression);
-  CGUIInfoLabel::ReplaceSpecialKeywordReferences(work, "EXP", [&](const std::string &str) -> std::string {
-    std::map<std::string, std::string>::const_iterator it = m_expressions.find(str);
-    if (it != m_expressions.end())
-      return it->second;
+  std::string work(strInput);
+  CGUIInfoLabel::ReplaceSpecialKeywordReferences(work, "EXP", [&](const std::string &expression) -> std::string {
+    if (std::find(resolved.begin(), resolved.end(), expression) != resolved.end())
+      CLog::Log(LOGERROR, "Skin has a circular expression \"%s\": %s", resolved.back().c_str(), strInput.c_str());
```

Comment:
I would stick a `return std::string()` here and avoid the else clause and decrease nesting a bit.